### DISCUSSION TITLE
`gleam new .` should not panic if there's already a README.md in the dir

### DISCUSSION THREAD

**Discussion by raquentin** [gleam-lang/gleam#2990] [Ideas & suggestions]
I think it's a pretty common workflow to create a repo on github w/ readme and perhaps license and then actually initialize the project locally.

While I do think that this should just skip over files like README.md and gitignore by default, there should at least be a more generic `--skip <FILENAME>` option so you can actually run `gleam new` without deleting the readme and then deleting the stuff in the generic Gleam one just to replace it with what was there to begin with.

Thoughts?

**Comment by lpil** [comment]
Let's not skip by default as we don't want to lose data by accident. `--skip README.md` is longer than `rm README.md` and the programmer is less likely to know that it needs to be done. What other options are there?

**Reply by raquentin** [reply]
I agree about not losing data, I'm talking about the case where a user intentionally has their own `README.md` and they run `gleam new .`, which panics. So they must `mv README.md README.md.tmp` just to run `gleam new .` then `rm -rf README.md` then `mv README.md.tmp README.md`, and same with gitignore. Obviously it's not super bothersome, but perhaps could be a bit more ergonomic to have some sort of message in the panic referencing a command `--skip` or similar which does not try to overwrite existing `README` or `gitignore` nor panic at their existence.

I'd say the options are:
- Write Gleam's readme and gitignore to gleam.readme.md and gleam.gitignore.md in the case that a dir already has a readme or gitignore
- Keep it as is
- Add `--skip` option to initialize gleam project without panicking at existing readme
- Some more that involve command line menus, likely unfavorable

Looking back I think the `gleam.` prefix option is my favorite as it avoids panic while still allowing the user to choose whether their initial `README.md` was there intentionally or not.

**Comment by han-tyumi** [comment]
What I've commonly seen with these types of commands is that by default it will never overwrite any existing files.
But a flag, such as `--force` & `-f`, will allow overwriting similar to `rm -f`.

I also like when the command appends to a file like gitignore if it already exists.

In general I think not doing anything destructive by default is a good idea.

**Comment by spenserblack** [comment]
In addition to allowing the README.md to exist, I think `gleam new` should *append* to the `.gitignore` rather than fail if it exists. For example, this is the current behavior of `cargo init`.

I'm not sure if `gleam new .` should skip an existing README.md, or append to it. Either are okay IMO, but I agree that requiring the developer to delete/rename/stash it, then restore it, is cumbersome.
